cryptography/src/commonMain/kotlin/com/wire/kalium/cryptography/backup/Backup.kt

```diff
@@ -0,0 +1,147 @@
+package com.wire.kalium.cryptography.backup
+
+import com.ionspin.kotlin.crypto.pwhash.PasswordHash
+import com.ionspin.kotlin.crypto.pwhash.crypto_pwhash_ALG_DEFAULT
+import com.ionspin.kotlin.crypto.pwhash.crypto_pwhash_SALTBYTES
+import com.wire.kalium.cryptography.CryptoUserID
+import com.wire.kalium.cryptography.utils.initializeLibsodiumIfNeeded
+import okio.Buffer
+import okio.IOException
+import okio.Source
+
+@OptIn(ExperimentalUnsignedTypes::class)
+class Backup(val salt: UByteArray, val userId: CryptoUserID, val passphrase: Passphrase) {
+
+    suspend fun provideHeaderBuffer(): ByteArray {
+        val hashedUserId = hashUserId(userId, salt)
+        return Header(format, version, salt, hashedUserId, OPSLIMIT_INTERACTIVE_VALUE, MEMLIMIT_INTERACTIVE_VALUE).toByteArray()
+    }
+
+    data class Passphrase(val password: String)
```

Comment:
I would move this out of the Backup class if `Backup` becomes an internal class.